### ISSUE TITLE
[Bug] typeDecl.derivedTypeDeclTransitive not work

### ISSUE THREAD

**Issue by evilpan** [joernio/joern#2875]
**Describe the bug**
While using joern to query an Android application, `derivedTypeDeclTransitive` returns empty result:
```
joern> cpg.typeDecl.fullNameExact("androidx.appcompat.app.AppCompatActivity").derivedTypeDeclTransitive.l
res41: List[TypeDecl] = List()
```

However `inheritsFromTypeFullName` got the results:
```
joern> cpg.typeDecl.filter(_.inheritsFromTypeFullName.contains("androidx.appcompat.app.AppCompatActivity")).fullName.l
res40: List[String] = List(
  "oversecured.ovaa.activities.EntranceActivity",
  "oversecured.ovaa.activities.MainActivity",
  "oversecured.ovaa.activities.WebViewActivity",
  "oversecured.ovaa.activities.LoginActivity",
  "oversecured.ovaa.activities.DeeplinkActivity"
)
```

**To Reproduce**
1. joern openCode("ovaa.apk")
2. perform query

the APK source can be found at https://github.com/oversecured/ovaa, And the the binary file is [ovaa.apk.zip](https://github.com/joernio/joern/files/11744873/ovaa.apk.zip)

**Expected behavior**
Get list of activities like the second query.

**Desktop (please complete the following information):**
 - OS: Ubuntu 22.04.1 LTS (Jammy Jellyfish)
 - Joern Version: 1.2.12+0-6ff8c800+20230611-1839
 - Java version: OpenJDK Runtime Environment (build 17.0.7+7-Ubuntu-0ubuntu122.04.2)

**Comment by evilpan** [comment]
I generate the cpg with following command:
```
jimple2cpg --android /path/to/android.jar -o ovaa.cpg ovaa.apk
```

And then use `importCpg` in joern to import the `ovaa.cpg`.

**Comment by prabhu** [comment]
<img width="1410" alt="2023-06-16_12-47-56" src="https://github.com/joernio/joern/assets/7842/06a3948f-a897-4684-8b7a-7a8c6721bdda">

**Comment by evilpan** [comment]
@prabhu I'm using the exact same SDK version, aka `$ANDROID_HOME/platforms/android-33/android.jar`. However the joern I use is built from source. That's strange, I'll update the code and test with the release version to see if get different results.

**Comment by prabhu** [comment]
Could you compare the difference in jimple2cpg between the two versions? I made some changes recently, but it might not cause any difference since those were optional flags.

**Comment by evilpan** [comment]
@prabhu After updating to the latest commit it works, however I did not find any changes in jimple2cpg frontend, which the diff comparation is https://github.com/joernio/joern/compare/6ff8c8009..58358b433

**Comment by evilpan** [comment]
Update:
It works in the first time. But later when I import it from saved workspace, it shows empty result again!

```
joern> workspace
res0: workspacehandling.WorkspaceManager[io.joern.joerncli.console.JoernProject] =
___________________________________________________________________________________________________________
| name    | overlays                                       | inputPath                             | open  |
|==========================================================================================================|
| cpg.bin | callgraph,controlflow,typerel,base,dataflowOss | /tmp/aaf/oversecured.ovaa_1.0/cpg.bin | false |

joern> workspace.openProject("cpg.bin")
Creating working copy of CPG to be safe
Loading base CPG from: /tmp/aaf/oversecured.ovaa_1.0/workspace/cpg.bin/cpg.bin.tmp
res1: Option[workspacehandling.Project] = Some(
  value = Project(
    projectFile = ProjectFile(inputPath = "/tmp/aaf/oversecured.ovaa_1.0/cpg.bin", name = "cpg.bin"),
    path = /tmp/aaf/oversecured.ovaa_1.0/workspace/cpg.bin,
    cpg = Some(value = Cpg (Graph [846091 nodes]))
  )
)

joern> cpg.typeDecl.fullNameExact("androidx.appcompat.app.AppCompatActivity").derivedTypeDeclTransitive.fullName.l
res2: List[String] = List()
```

Note that while opening from workspace, the cpg only contains *846091* nodes, but if you use `importCpg` it will be *926362* nodes instead.

According to my test, the reason why the saved workspace miss some changes is that I **did not** enter `save` command    explicitly after first importCpg. The prompt of joern says:

```
The graph has been modified. You may want to use the `save` command to persist changes to disk.  All changes will also be saved collectively on exit
```

But seems that the changes are not automatically saved on exit.

So in conclusion, this issue is caused by not correctly saved cpg in workspace, maybe I didn't understand it correctly.
